inc/GitHubUpdater.php

```diff
@@ -49,19 +49,101 @@ public function __construct( GP_Project $project ) {
 	}
 
 	/**
-	 * Returns the repository URL based on the project's source URL template.
+	 * Returns the name for a Git repository.
 	 *
-	 * @since 2.0.0
+	 * @since 2.0.3
 	 *
-	 * @return string SSH URL to the repository, e.g. git@github.com:wearerequired/required-valencia.git.
+	 * @return string Repository name in the form <owner>/<repo>, e.g. wearerequired/traduttore.
 	 */
-	protected function get_ssh_url(): string {
-		// e.g. https://github.com/wearerequired/required-valencia/blob/master/%file%#L%line%.
+	protected function get_repository_name() : string {
+		// e.g. https://github.com/wearerequired/traduttore/blob/master/%file%#L%line%.
 		$url   = $this->project->source_url_template();
 		$parts = explode( '/blob/', wp_parse_url( $url, PHP_URL_PATH ) );
 		$path  = array_shift( $parts );
 
-		return sprintf( 'git@github.com:%s.git', ltrim( $path, '/' ) );
+		return ltrim( $path, '/' );
+	}
+
+	/**
+	 * Indicates whether a GitHub repository is publicly accessible or not.
+	 *
+	 * @since 2.0.3.
+	 *
+	 * @return bool Whether the repository is publicly accessible.
+	 */
+	protected function is_public_repository() : bool {
+		$response = wp_remote_head( 'https://api.github.com/repos/' . $this->get_repository_name() );
+
+		return 200 === wp_remote_retrieve_response_code( $response );
+	}
+
+	/**
+	 * Returns the repository's clone URL.
+	 *
+	 * Supports either HTTPS or SSH.
+	 *
+	 * @since 2.0.3
+	 *
+	 * @return string SSH URL to the repository, e.g. git@github.com:wearerequired/traduttore.git
+	 *                or https://github.com/wearerequired/traduttore.git.
+	 */
+	protected function get_clone_url() : string {
+		/**
+		 * Filters whether HTTPS or SSH should be used to clone a repository.
+		 *
+		 * @since 2.0.3
+		 *
+		 * @param bool       $use_https Whether to use HTTPS or SSH. Defaults to HTTPS for public repositories.
+		 * @param GP_Project $project   The currrent GlotPress project.
+		 */
+		$use_https = apply_filters( 'traduttore_git_clone_use_https', $this->is_public_repository(), $this->project );
+
+		$clone_url = $this->get_ssh_url();
```

Comment:
This should be in the `else` branch since `get_ssh_url()` does to much, like also calling `get_repository_name()`.

Reply:
> `like also calling get_repository_name().``

So does `get_https_url()`. If something does too much, it's `get_https_url()`, not ``get_ssh_url()`.

`get_repository_name()` is just a super tiny helper function to get the `wearerequired/traduttore` part out of a repo URL.

Reply:
Right, because `get_https_url()` is calling it too we should ensure it's only called once in this condition.

I don't see `get_repository_name()` as "super tiny". `GP_Project::source_url_template()` itself has a lot of logic, then there's `wp_parse_url()` + `explode()` + `array_shift()` and finally `ltrim()`.

Have you considered making the repository name a property which is initialized in the class constructor as it depends on the project which can't be changed afterwards?

Reply:
I have considered that, but put it on the "we can do that later" pile.